test/hotspot/jtreg/serviceability/jvmti/vthread/VThreadEventTest/VThreadEventTest.java

```diff
@@ -0,0 +1,180 @@
+    static final long TIMEOUT_BASE = 1_000_000L;
+
+    private static void log(String msg) { System.out.println(msg); }
+
+    private static native int threadEndCount();
+    private static native int threadMountCount();
+    private static native int threadUnmountCount();
+
+    private static volatile int completedNo;
+    private static volatile boolean attached;
+    private static boolean failed;
+    private static List<Thread> threads = new ArrayList(TCNT1);
+
+    public static void main(String[] args) throws Exception {
+        if (Runtime.getRuntime().availableProcessors() < 8) {
+            log("WARNING: test expects at least 8 processors.");
+        }
+        Counter ready1 = new Counter(THREAD_CNT);
+        Counter ready2 = new Counter(THREAD_CNT);
+        Counter mready = new Counter(1);
+
+        try (ExecutorService executorService = Executors.newVirtualThreadPerTaskExecutor()) {
+            for (int tCnt = 0; tCnt < TCNT1; tCnt++) {
+                executorService.execute(() -> {
+                    synchronized (threads) {
+                        threads.add(Thread.currentThread());
+                    }
+                    log("test1 vthread started");
+                    ready1.decr();
+                    mready.await();
+                    try {
+                        // timeout is big enough to keep mounted untill interrupted
+                        Thread.sleep(20000);
+                    } catch (InterruptedException ex) {
+                        // it is expected, ignore
+                    }
+                    ready2.decr();
+                    completedNo++;
+                });
+            }
+            for (int tCnt = 0; tCnt < TCNT2; tCnt++) {
+                executorService.execute(() -> {
+                    log("test2 vthread started");
+                    ready1.decr();
+                    mready.await();
+                    while (!attached) {
+                        // keep mounted
+                    }
+                    ready2.decr();
+                    completedNo++;
+                });
+            }
+            for (int tCnt = 0; tCnt < TCNT3; tCnt++) {
+                executorService.execute(() -> {
+                    log("test3 vthread started");
+                    ready1.decr();
+                    mready.await();
+                    while (!attached) {
+                        // keep mounted
+                    }
+                    LockSupport.parkNanos(10 * TIMEOUT_BASE);
+                    ready2.decr();
+                    completedNo++;
+                });
+            }
+            ready1.await();
+            mready.decr();
+            VirtualMachine vm = VirtualMachine.attach(String.valueOf(ProcessHandle.current().pid()));
+            vm.loadAgentLibrary("VThreadEventTest");
+            Thread.sleep(100);
+            log("main: completedNo: " + completedNo);
+            attached = true;
+            for (Thread t : threads) {
```

Comment:
AFAIU threads in 3rd group (TCNT3) should be unmounted (with  LockSupport.parkNanos) before they are interrupted.
Then we need sleep here

Reply:
Not sure, I understand the  suggestion.
The only interrupted threads are those in the TCN1 group as only these threads are added to the threads list.
Please, see the line 95.
Do you still think an extra timeout is needed here?